File: app/src/ultimate/java/com/clockmods/widget/model/WidgetThemeSpec.java
package com.clockmods.widget.model;
public final class WidgetThemeSpec {
    public enum Font { SANS, SERIF, MONOSPACE }
    public final String id;
    public final int displayNameRes, backgroundDrawableRes, primaryTextColor, secondaryTextColor, accentColor, iconColor, defaultBackgroundAlpha;
    public final Font fontLayoutVariant;
    public final boolean supportsAnalog = true, supportsDigital = true, supportsWeather = true, supportsCalendar = true;
    public WidgetThemeSpec(String id, int name, int background, int primary, int secondary, int accent, Font font, int alpha) {
        this.id=id; displayNameRes=name; backgroundDrawableRes=background; primaryTextColor=primary;
        secondaryTextColor=secondary; accentColor=accent; iconColor=accent; fontLayoutVariant=font; defaultBackgroundAlpha=alpha;
    }
}

File: app/src/ultimate/java/com/clockmods/widget/model/WidgetThemeSpec.kt
package com.clockmods.widget.model

class WidgetThemeSpec(
    @JvmField val id: String,
    @JvmField val displayNameRes: Int,
    @JvmField val backgroundDrawableRes: Int,
    @JvmField val primaryTextColor: Int,
    @JvmField val secondaryTextColor: Int,
    @JvmField val accentColor: Int,
    @JvmField val fontLayoutVariant: Font,
    @JvmField val defaultBackgroundAlpha: Int,
) {
    enum class Font { SANS, SERIF, MONOSPACE }

    @JvmField val iconColor: Int = accentColor
    @JvmField val supportsAnalog: Boolean = true
    @JvmField val supportsDigital: Boolean = true
    @JvmField val supportsWeather: Boolean = true
    @JvmField val supportsCalendar: Boolean = true
}
